### PR TITLE
Line-buffered output for %%magic scripts

### PR DESCRIPTION
**Pull request by jetpacktuxedo**
This was done to address #12595

The following will now behave more like what you would expect:
```bash
In [1]: %%bash
   ...: for i in 1 2 3; do
   ...:   echo $i
   ...:   sleep $1
   ...: done
```

This doesn't actually _fully_ address it, as it is still line buffered. The following will still exhibit behavior more like before:
```bash
In [1]: %%bash
   ...: for i in 1 2 3; do
   ...:   printf $i
   ...:   sleep $1
   ...: done
```

I also seem to have some test failures right now related to the event loop being closed unexpectedly. Hopefully I can get those sorted shortly, but I wanted to go ahead and open this up to try to get some feedback on whether or not this is even worth pursuing.

### CODE REVIEW

File: IPython/core/magics/script.py [hunk 1]
```diff
@@ -8,9 +8,15 @@
...
 import time
-from subprocess import Popen, PIPE, CalledProcessError
+import asyncio
 import atexit
 
+try:
+    from asyncio import create_task
+except ImportError:
+    # 3.6 had this function in a different module, this can go away when 3.6 is dropped
+    from AbstractEventLoop import create_task
```
Comment (jetpacktuxedo) [inline IPython/core/magics/script.py:18]: 3.6 isn't supported beyond IPython 7.16 so I guess I can drop all of this? I'll have test failures in travis, though...

**Comment by jetpacktuxedo** [comment]
Alright, I think this is totally ready now. The only travis failure is 3.6 which is no longer supported on this branch (should probably be dropped from travis...).

File: IPython/core/magics/script.py [hunk 2]
```diff
@@ -229,20 +262,9 @@ def shebang(self, line, cell):
...
-            sys.stdout.write(out)
-            sys.stdout.flush()
-        if args.err:
-            self.shell.user_ns[args.err] = err
-        else:
-            sys.stderr.write(err)
-            sys.stderr.flush()
         if args.raise_error and p.returncode!=0:
-            raise CalledProcessError(p.returncode, cell, output=out, stderr=err)
+            print(p.returncode)
```
Comment (Carreau) [inline IPython/core/magics/script.py:266]: missing debug ?

**Comment by Carreau** [comment]
> Alright, I think this is totally ready now. The only travis failure is 3.6 which is no longer supported on this branch (should probably be dropped from travis...).

@SylvainCorlay I think I kept 3.6 testing for you and Xeus, do you think we can drop 3.6 now on master (7.x will still test on 3.6).

**Comment by SylvainCorlay** [comment]
That would make sense to drop 3.6.

**Comment by SylvainCorlay** [comment]
@johanmabille do you recall how to fix this MT/MD issue with gtest?

**Comment by JohanMabille** [comment]
I think we replaced the flags with something like

```
set(CompilerFlags
        CMAKE_CXX_FLAGS
        CMAKE_CXX_FLAGS_DEBUG
        CMAKE_CXX_FLAGS_RELEASE
        CMAKE_C_FLAGS
        CMAKE_C_FLAGS_DEBUG
        CMAKE_C_FLAGS_RELEASE
        )
foreach(CompilerFlag ${CompilerFlags})
  string(REPLACE "/MD" "/MT" ${CompilerFlag} "${${CompilerFlag}}")
endforeach()
```

File: IPython/core/magics/script.py (hunk 2 above)
Comment (jetpacktuxedo) [reply IPython/core/magics/script.py:266]: Whoops, that might be something I flipped when I was debugging and forgot to out back. Lemme take a look.

**Comment by SylvainCorlay** [comment]
Woops, mixed things up with another thread!

**Comment by Carreau** [comment]
Thanks, no worry for the off-topic, I've hidden the messages.

**Comment by Carreau** [comment]
I pushed a commit that remove testing on 3.6

**Comment by jetpacktuxedo** [comment]
I rebased, but travis still seems to be running tests on 3.6 🤔

**Comment by Carreau** [comment]
> I pushed a commit that remove testing on 3.6

ha, you seem to have overwritten the commit I pushed on this branch, I'll redo it.

**Comment by Carreau** [comment]
I just pushed on your branch (this PR) 958d1d1 that removes testing on 3.6

**Comment by jetpacktuxedo** [comment]
> > I pushed a commit that remove testing on 3.6
> ha, you seem to have overwritten the commit I pushed on this branch, I'll redo it.

Oh, yep, I probably did when I rebased 😅

**Comment by Carreau** [comment]
let's try.

**Pull request merged**